pints/_mcmc/_emcee_hammer.py

```diff
@@ -82,15 +82,13 @@ def ask(self):
             self._remaining = np.delete(self._remaining, 0)
 
             # Pick j from the complementary ensemble
-            j = np.random.randint(self._n_chains - 1)
-            if j >= self._k:
-                j += 1
+            compl_indices = list(set(range(self._n_chains)) -
+                                 set([self._k]))
+            j = np.random.choice(compl_indices)
```

Comment:
Sorry I don't know much about this sampler. 
- k is the index of the current runner/chain?
- j is the new runner that you want to pick?

Would it potentially be easier (and at least for me easier to understand) if you did something like:
```
# Create list of runners (Could also be done in the init)
self._runners = list(range(self._n_chains)) 

# Pick free runner
free_runners = self._runners.copy()
free_runners.remove(self._k)
j = np.random.choice(free_runners)
```

Reply:
Good point: I've created another method that handles this and uses the code you've posted. Re: k and j, yes, that's right: I know j and k aren't informative but they match the notation in the paper, so I'd like to keep them like that.